reanahub/reana-workflow-engine-serial
# add kubernetes_job_timeout parameter

closes reanahub/reana-job-controller#343

How to test:

This PR is part of a bigger group of PRs. Please refer to the addressed issue to see all of them.

1. Modify Serial helloworld example, with `kubernetes_job_timeout=20` (in seconds) and `sleeptime=2` (2 seconds so the workflow will definitely run longer than 20 seconds):

`reana.yml`
```yaml
...
parameters:
    helloworld: code/helloworld.py
    inputfile: data/names.txt
    outputfile: results/greetings.txt
    sleeptime: 2  # <= here 
...
steps:
      - environment: 'python:2.7-slim'
        kubernetes_job_timeout: 20  # <= here
...
```

2. Start workflow `reana-client run -w serial-timeout -f reana.yaml`

3. After 25-30 seconds, check using `reana-client list` if workflow failed. It should.

4. Using `reana-client logs` check if the reason is `Job was killed due to timeout.`

5. Try to input string or object instead of integer for `kubernetes_job_timeout`, the workflow should fail with an error message like:

```
('Error while marshalling value=not_correct_timeout to type=integer/int32.', ValueError("invalid literal for int() with base 10: 'not_correct_timeout'"))
```

Check it using `reana-client logs`.

**Note:** You will see an error message above, but you will not see engine logs in the `reana-client logs` output. This is due to [this issue](https://github.com/reanahub/reana-workflow-controller/issues/426). It will be fixed after this PR.
